jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java

```diff
@@ -632,7 +633,7 @@ private void setCatchEventProperties(CatchEvent event,
             for (DataOutputAssociation doa : outputAssociations) {
                 String doaName = ((DataOutput) doa.getSourceRef().get(0)).getName();
                 if (doaName != null && doaName.length() > 0) {
-                    doutassociationbuff.append("[dout]" + ((DataOutput) doa.getSourceRef().get(0)).getName());
+                    doutassociationbuff.append("[dout]" + updateDataInputOutputUnderscores(((DataOutput) doa.getSourceRef().get(0)).getName()));
```

Comment:
@tsurdilo I had a look on all places in this class where `[din]` or `[dout]` is constructed. I am not sure if it is correct, because each construction of `[din]` or `[dout]` uses methods `updateDataInputOutputUnderscores` and `encodeAssociationValue` in slightly different manner. Could you please double check this? Or at least could you give me some explanation how it should work please?

Reply:
@jomarko for assignments we encode the value of the assignment so it does not break the xml. For associations like this one we replace just the spaces to underscores because of this jira (for dmn spaces in data input/output names are required but since we use their name to assemble the id of the data input/output/itemdefinition we have to replace them to underscores).